File: problems/0057_insert_interval.cpp
using namespace std;
#import <vector>

class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        vector<vector<int>> answer;
        for(int i = 0; i < intervals.size(); ++i)
        {
            if(intervals[i][1] < newInterval[0])
            {
                answer.push_back(intervals[i]);
            }

            else if(intervals[i][0] > newInterval[1])
            {
                answer.push_back(newInterval);
                newInterval = intervals[i];
            }

            else if(intervals[i][1] >= newInterval[0] || intervals[i][0] <= newInterval[1])
            {
                newInterval[0] = std::min(newInterval[0], intervals[i][0]);
                newInterval[1] = std::max(newInterval[1], intervals[i][1]);
            }

        }
        answer.push_back(newInterval);
        return answer;
    }
};

File: problems/0057_insert_interval.go
package main

func Min(i int, j int) int {
	if i < j {
		return i
	}
	return j
}

func Max(i int, j int) int {
	if i > j {
		return i
	}
	return j
}

func insert(intervals [][]int, newInterval []int) [][]int {
	answer := make([][]int, 0)
	for i := 0; i < len(intervals); i++ {
		if intervals[i][1] < newInterval[0] {
			answer = append(answer, intervals[i])
		} else if intervals[i][0] > newInterval[1] {
			answer = append(answer, newInterval)
			newInterval = intervals[i]
		} else if intervals[i][1] >= newInterval[0] || intervals[i][0] <= newInterval[1] {

			newInterval[0] = Min(newInterval[0], intervals[i][0])
			newInterval[1] = Max(newInterval[1], intervals[i][1])
		}
	}
	answer = append(answer, newInterval)

	return answer
}
